src/parser/index.ts

```diff
@@ -145,31 +145,46 @@ export class Parser {
 
     /**
      * Simplify the logical expression if possible.
+     * - AND, OR with one operand is collapsed.
+     * - Logical expressions without operands are treated as collection.
+     * @param collapsible
      * @param operands
      */
-    const logicalExpressionReducer = (operands: Evaluable[]): Evaluable | undefined =>
-      operands.length === 1 ? operands[0] : undefined
+    const logicalExpressionReducer = (
+      collapsible: boolean,
+      operands: Evaluable[]
+    ): Evaluable | undefined => {
+      if (operands.length === 0) {
+        return this.getOperand(raw)
+      }
+      return collapsible && operands.length === 1
+        ? operands[0]
+        : undefined
+    }
```

Comment:
Does this account for this case:
```json
["IN", "$var", ["OR", "VT"]]
```

Reply:
It should now, a2f8e09c9ebee85ee86678853af369908227ae0b